src/datasources/config-api/config-api.service.ts

```diff
@@ -1,32 +1,62 @@
 import { Injectable } from '@nestjs/common';
-import { Page } from './entities/page.entity';
-import { Chain } from './entities/chain.entity';
 import { Inject } from '@nestjs/common';
 import { IConfigurationService } from '../../common/config/configuration.service.interface';
 import { CacheFirstDataSource } from '../cache/cache.first.data.source';
+import { Page } from '../../common/entities/page.entity';
+import { Chain } from '../../chains/entities';
+import { DefinedError, ValidateFunction } from 'ajv';
+import { ValidationErrorFactory } from '../errors/validation-error-factory';
+import { JsonSchemaService } from '../common/json-schema.service';
+import {
+  chainSchema,
+  nativeCurrencySchema,
+} from '../../chains/entities/schemas/chain.schema';
```

Comment:
This introduces a dependency between the datasource and a route.

We should have a schema for the Safe Config Service and a schema for the Safe Transaction Service. Additionally we would have a schema for the Safe Client Gateway if we want to validate the responses returned by this service.

I'd keep these schemas separate. If they do match (external service + cgw) we can add it to `/common`.

The `routes` should depend on the `datasources`. `Datasources` should depend on internal components (cache, networking). `/common` is still a baseline that can be used by both at this point. Ideally we'd have a common module for the routes only and a common module for the data sources only for better encapsulation.

Reply:
Fixed in [84c402f](https://github.com/5afe/safe-client-gateway-nest/commit/84c402fcafe4e82ec8817e0092b9fd35427ec1a3) (I moved the schema to the `datasources` layer as we discussed, maybe we could give this a second thought when we take a final decision on layers architecture as we discussed 🙂)